### ISSUE TITLE
mypy accepts tuple[int] etc.

### ISSUE THREAD

**Issue by ilevkivskyi** [python/mypy#2428]
mypy currently accepts
```python
tuple[int], list[int], dict[int, str]
```
while all of those fail at runtime. I think those should be flagged as errors.

**Comment by JukkaL** [comment]
This is an old bug, but I couldn't find an existing issue. #999 is closely related -- fixing either of these issues would likely make the remaining issue trivial to fix.
